llama_index/indices/base.py

```diff
@@ -189,23 +189,61 @@ def insert(self, document: Document, **insert_kwargs: Any) -> None:
 
     @abstractmethod
     def _delete(self, doc_id: str, **delete_kwargs: Any) -> None:
-        """Delete a document."""
+        """Delete a node."""
+
+    def delete_nodes(
+        self,
+        doc_ids: List[str],
+        delete_from_docstore: bool = False,
+        **delete_kwargs: Any,
+    ) -> None:
+        """Delete a list of nodes from the index.
+
+        Args:
+            doc_ids (List[str]): A list of doc_ids from the nodes to delete
+
+        """
+        for doc_id in doc_ids:
+            self._delete(doc_id, **delete_kwargs)
+            if delete_from_docstore:
+                self.docstore.delete_document(doc_id, raise_error=False)
+
+        self._storage_context.index_store.add_index_struct(self._index_struct)
 
     def delete(self, doc_id: str, **delete_kwargs: Any) -> None:
         """Delete a document from the index.
-
         All nodes in the index related to the index will be deleted.
 
         Args:
-            doc_id (str): document id
+            doc_id (str): A doc_id of the ingested document
 
         """
-        logger.debug(f"> Deleting document: {doc_id}")
-        self._delete(doc_id, **delete_kwargs)
-        self._storage_context.index_store.add_index_struct(self._index_struct)
+        logger.warning(
+            "delete() is now deprecated, please refer to delete_ref_doc() to delete "
+            "ingested documents+nodes or delete_nodes to delete a list of nodes."
+        )
+        self.delete_ref_doc(doc_id)
+
+    def delete_ref_doc(
```

Comment:
are we committing to calling raw documents `ref_doc` in contrast to nodes?

I think it's an improvement from before, but there's still a lot of confusion created by:
1. Nodes and Documents are both `BaseDocument`
2. we use `doc_id` to identify both `Node` and `Document`
3. We use `ref_doc_id` to refer to a `Document` in the context of a `Node`

Reply:
Not for this PR, but I'd like us to align on clearer abstractions and better naming.

Reply:
This is correct on all 4 counts. 

There are definitely better changes to be made, but I was trying my best to avoid breaking changes

I can go ham renaming stuff, but was unsure how drastically to change things

Reply:
Yep, not suggesting that we should go ham. Just writing this out for future reference.

Reply:
Now I'm starting to think it's okay to keep `delete` instead of renaming to `delete_ref_doc`. And just make the other operations on `node` more explicitly named.

The reason is that we have other index updating functions like `update` / `refresh` which all operate on raw documents, but don't have this `ref_doc` suffix right now. So I wonder if it'd be confusing.

Reply:
I can make update/refresh names have the ref_doc suffix maybe: `update_ref_doc` and `refresh_ref_docs` maybe. Would keep the naming a bit more consistent around the `ref_doc` term

Reply:
@Disiok yeah as @logan-markewich said, what about just deprecating `delete`, `update`, and `refresh` in favor of more precise names, but still keeping them around for backwards compat purposes?

Reply:
Sounds good.